### PR TITLE
Improve HTML encode to handle single quotes

### PR DESCRIPTION
**Pull request by slang25**
This library is great! I found that it doesn't HTML escape single quotes, so here's a small PR...

### CODE REVIEW

File: src/RazorBlade.Library/HtmlHelper.cs [hunk 1]
```diff
@@ -51,6 +51,7 @@ public string Encode(object? value)
                 '<'   => "&lt;",
                 '>'   => "&gt;",
                 '"'   => "&quot;",
+                "\'"  => "&#39;",
```
Comment (ltrzesniewski) [inline src/RazorBlade.Library/HtmlHelper.cs]: - That would need to be `'\''` (match on a char instead of a string)
- Why `&#39;` instead of `&apos;`?

**Comment by ltrzesniewski** [comment]
Thanks! I thought single quotes didn't really need to be escaped. Did you find a case where it's an issue?

**Review by ltrzesniewski — COMMENTED** [review]
There are also other places in the code where HTML escaping is performed. Those would need to be updated as well.

**Comment by slang25** [comment]
So I coded this up late last night, hence the mistakes 😄 
What led me to create this PR is that my test started failing after switching from RazorLight, it is expecting `&#x27;` for `'`.
From some research, it should be encoded to either `&#39;` or `&#x27;` (this is the hex representation), however I chose the hex implementation as it's what the other razor libraries do, and that's what the `System.Text.Encodings.Web.HtmlEncoder.Default` implementation does:
```csharp
System.Text.Encodings.Web.HtmlEncoder.Default.Encode("foo's").ShouldBe("foo&#x27;s");
```
So copying this would lead to less surprises for future adopters.

**Comment by ltrzesniewski** [comment]
Ok, thanks for the explanation. I'll review this in more detail tonight, but I can already say that:

- You didn't implement the `#else` case in `HtmlHelper`
- `HtmlTemplate.Write` needs the same changes

**Comment by slang25** [comment]
Thanks for catching those @ltrzesniewski, changes made

File: src/RazorBlade.Library/HtmlTemplate.cs [hunk 2]
```diff
@@ -47,7 +47,7 @@ protected internal override void Write(object? value)
 
         while (true)
         {
-            var idx = valueSpan.IndexOfAny("&<>\"");
+            var idx = valueSpan.IndexOfAny("&<>\"`'");
```
Comment (ltrzesniewski) [inline src/RazorBlade.Library/HtmlTemplate.cs]: That backtick looks wrong

Comment (slang25) [reply src/RazorBlade.Library/HtmlTemplate.cs]: I have no excuse this time 🤦‍♂️

**Comment by slang25** [comment]
I'm thinking this should also handle chars outside of the ASCII range, let me take a quick look at that too

**Comment by ltrzesniewski** [comment]
I don't think that's really required, everything handles Unicode now.

**Comment by ltrzesniewski** [comment]
I released this in v0.4.3 - thanks! 🙂

**Comment by slang25** [comment]
Amazing, thanks @ltrzesniewski

**Pull request merged**